DatadogCore/Sources/Datadog.swift

```diff
@@ -231,7 +231,14 @@ public enum Datadog {
     /// Verbosity level of Datadog SDK. Can be used for debugging purposes.
     /// If set, internal events occuring inside SDK will be printed to debugger console if their level is equal or greater than `verbosityLevel`.
     /// Default is `nil`.
-    public static var verbosityLevel: CoreLoggerLevel? = nil
+    public static var verbosityLevel: CoreLoggerLevel? {
+        get { _verbosityLevel.wrappedValue }
+        set { _verbosityLevel.wrappedValue = newValue }
+    }
+
+    /// The backing storage for `verbosityLevel`, ensuring efficient synchronized
+    /// read/write access to the shared value.
+    private static let _verbosityLevel = ReadWriteLock<CoreLoggerLevel?>(wrappedValue: nil)
```

Comment:
```suggestion
    @ReadWriteLock
    public static var verbosityLevel: CoreLoggerLevel? = nil
```

I wonder if you really need a backing property here.

Reply:
Unfortunately, property wrappers are unusable here; the Swift 6 / `complete` concurrency checker sees this as mutable state:

<img width="874" alt="Screenshot 2024-04-17 at 7 19 21 AM" src="https://github.com/DataDog/dd-sdk-ios/assets/720600/7fb64360-61d7-4656-9c58-e8cfc53bfc34">

These two requirements are at odds:

- Property wrappers must be defined as `var` not `let`.
- The concurrency checker disallows the use of `var` because it's shared mutable state

Something has to give. Until the day that Swift gains the ability to overlook a `var` wrapped in a sendable property wrapper, it is not possible to use a property wrapper to resolve concurrency issues with shared mutable state.